dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java

```diff
@@ -297,33 +299,58 @@ protected boolean rewriteDockerfile(String img, String tag,
      * If the inbound {@code line} does not qualify for changes or if the tag is already correct, the
      * {@code stringBuilder} will get {@code line} added to it. We return {@code false} in this instance.
      *
+     *
+     * @param reader BufferReader used for reading next line to FROM Instruction
      * @param imageToFind the Docker image that may require a tag update
      * @param tag the Docker tag that we'd like the image to have
      * @param stringBuilder the stringBuilder to accumulate the output lines for the pull request
      * @param line the inbound line from the Dockerfile
      * @return Whether we've modified the {@code line} that goes into {@code stringBuilder}
      */
-    protected boolean changeIfDockerfileBaseImageLine(String imageToFind, String tag,
+    protected boolean changeIfDockerfileBaseImageLine(BufferedReader reader, String imageToFind, String tag,
                                                       StringBuilder stringBuilder, String line,
-                                                      String ignoreImageString) {
+                                                      String ignoreImageString) throws IOException {
         boolean modified = false;
         String outputLine = line;
+        String nextLine = null;
 
         // Only check/modify lines which contain a FROM instruction
         if (FromInstruction.isFromInstruction(line)) {
             FromInstruction fromInstruction = new FromInstruction(line);
+            nextLine = reader.readLine();
```

Comment:
Write a unit test containing the following:

```
FROM <some/image/path>
FROM <some/other/image/path>
#totally-not-no-dfiu
```
I'm pretty sure DFIU will be applied to just the first FROM statement.  The first FROM triggers the if on line 318.  The second  FROM doesn't pass the ignorePR test on line 323 so the first FROM Instruction is updated.  However, the second FROM instruction is already consumed from the buffered reader.  So on the next call to `changeIfDockerfileBaseImageLine` the line being read will be `#totally-not-no-dfiu` which will be ignored.  Lets test this.